tests/boulder-integration.sh

```diff
@@ -37,14 +37,41 @@ common() {
 # unrequested challenge to prevent regressions in #3601.
 python -m SimpleHTTPServer $http_01_port &
 python_server_pid=$!
-common --domains le1.wtf --preferred-challenges tls-sni-01 auth
+
+export HOOK_TEST="/tmp/hook$$"
+common --domains le1.wtf --preferred-challenges tls-sni-01 auth \
+       --pre-hook 'echo wtf.pre >> "$HOOK_TEST"' \
+       --post-hook 'echo wtf.post >> "$HOOK_TEST"'\
+       --renew-hook 'echo renew >> "$HOOK_TEST"'
 kill $python_server_pid
 python -m SimpleHTTPServer $tls_sni_01_port &
 python_server_pid=$!
-common --domains le2.wtf --preferred-challenges http-01 run
+common --domains le2.wtf --preferred-challenges http-01 run \
+       --pre-hook 'echo wtf.pre >> "$HOOK_TEST"' \
+       --post-hook 'echo wtf.post >> "$HOOK_TEST"'\
+       --renew-hook 'echo renew >> "$HOOK_TEST"'
 kill $python_server_pid
 
-common -a manual -d le.wtf auth --rsa-key-size 4096
+common -a manual -d le.wtf auth --rsa-key-size 4096 \
+       --pre-hook 'echo wtf2.pre >> "$HOOK_TEST"' \
+       --post-hook 'echo wtf2.post >> "$HOOK_TEST"'
+
+CheckHooks() {
+    EXPECTED="/tmp/expected$$"
+    echo "wtf.pre" > "$EXPECTED"
+    echo "wtf2.pre" >> "$EXPECTED"
+    echo "renew" >> "$EXPECTED"
+    echo "renew" >> "$EXPECTED"
+    echo "wtf.post" > "$EXPECTED"
+    echo "wtf2.post" >> "$EXPECTED"
+    if cmp --quiet "$EXPECTED" "$HOOK_TEST" ; then
```

Comment:
Three options:
1. We live with that risk, on the theory that we're probably the only people running these unit tests, and we'll fix thing if a future version of Python changes this
2. We wrap the `glob.glob` call in `sorted()`, causing our renewal order to be deterministic but subtly risking breaking anyone who assumed that our renewal order wouldn't change ;)
3. Remove the integration test, or write it in some crazy order-agnostic way.